docs/documentation/algorithms/fidex/fidex.md

```diff
@@ -0,0 +1,503 @@
+| JSON identifier         | `weights_file`      |
+| Default value           | `None`              |
+
+!!! warning
+    \*\*This argument is not required if, **and only if**, the [rules file](#rules-file) is specified instead. 
+
+--- 
+
+### Rules file
+*File containing a collection of rules.*
+
+|  **Property**           | **Value**           |
+|:------------------------|:--------------------|
+| Is required             | No**                |
+| Type                    | `String`            |
+| CLI argument syntax     | `--rules_file`      | 
+| JSON identifier         | `rules_file`        |
+| Default value           | `None`              |
+
+!!! warning
+    \*\*This argument is not required if, **and only if**, the [weights file](#weights-file) is specified instead. 
+
+--- 
+
+### Rules output file
+*Name of the file containing rules generated by `Fidex`.*
+
+|  **Property**           | **Value**           |
+|:------------------------|:--------------------|
+| Is required             | Yes                 |
+| Type                    | `String`            |
+| CLI argument syntax     | `--rules_outfile`   | 
+| JSON identifier         | `rules_outfile`     |
+| Default value           | `None`              |
+
+!!!info
+    The file name's extension can be specified as `.json`. Allowing the program to generate a JSON-structured rule output file.
+
+---
+
+### Number of attributes 
+*Number of attributes in the dataset (should be equal to the number of inputs of the model).*
+
+|  **Property**           | **Value**           |
+|:------------------------|:--------------------|
+| Is required             | Yes                 |
+| Type                    | `Integer`           |
+| CLI argument syntax     | `--nb_attributes`   | 
+| JSON identifier         | `nb_attributes`     |
+| Default value           | `None`              |
+
+---
+
+### Number of classes
+*Number of classes in the dataset (should be equal to the number of outputs of the model).*
+
+|  **Property**           | **Value**           |
+|:------------------------|:--------------------|
+| Is required             | Yes                 |
+| Type                    | `Integer`           |
+| CLI argument syntax     | `--nb_classes`      |
+| JSON identifier         | `nb_classes`        |
+| Default value           | `None`              |
+
+---
+
+### Test predictions file
+*File containing the predictions from the testing portion of the dataset.*
+
+|  **Property**           | **Value**           |
+|:------------------------|:--------------------|
+| Is required             | No**                |
+| Type                    | `String`           |
+| CLI argument syntax     | `--test_pred_file`  |
+| JSON identifier         | `test_pred_file`    |
+| Default value           | `None`              |
+
+!!!Warning
+    \*\*This argument is not required if, **and only if**, the predictions are already specified inside the [test data file](#test-data-file).  
```

Comment:
What is the prupose of this `\*\*`? Is this supposed to have a matching closing `\*\*`? If not, I'd suggest to remove it (the yellow warning box should be enough anyway).